services/ui-src/src/views/StateHome/index.tsx

```diff
@@ -287,6 +288,21 @@ const StateHome = () => {
       };
     });
 
+  //we are using the order of the coresetabbr enums to determine the order of the table items
+  const sortedTableItems: CoreSetTableItem.Data[] = [];
+  Object.values(CoreSetAbbr).forEach((abbr) => {
+    if (abbr === "HHCS") {
+      sortedTableItems.push(
+        ...formattedTableItems.filter((item) => item.coreSet.includes(abbr))
+      );
+      return;
+    }
+    const coreSetItem = formattedTableItems.find(
+      (item) => item.coreSet === abbr
+    );
+    if (coreSetItem) sortedTableItems.push(coreSetItem);
+  });
+
```

Comment:
Once again you've written a helpful comment - but if we were referencing `CoreSetDisplayOrder` instead of `Object.values(CoreSetAbbr)`, it would be unnecessary.

As always, I think `.forEach` is a less-than-ideal choice; `for..of` is superior in almost every way when creating a procedural loop. But in this case, I don't think we even want a procedural loop. Building up an array by pushing in elements one at a time, while referencing another array is the _exact_ time to use `.map`. The HHCS core sets add a small wrinkle we can iron out with `.flatMap` instead.

```suggestion
  const sortedTableItems = CoreSetDisplayOrder.flatMap((abbr) =>
   abbr === CoreSetAbbr.HHCS
      ? formattedTableItems.filter((item) => item.coreSet.startsWith(abbr))
      : formattedTableItems.filter((item) => item.coreSet === abbr)
  );

```

Reply:
i did something similar earlier on using map but it was less eloquent. On deploy it kept saying I had typescript issues, so I'm hoping that it was cause by me using enums and that its fine now that i've switched over to your code.